src/main/java/com/dnastack/gcp/client/DosClient.java

```diff
@@ -27,19 +27,26 @@
     private final HttpClient httpClient;
 
     private final Gson gson = new Gson();
+    private long objectId;
 
-    public DosClient(URI baseUrl, String username, String password) throws IOException {
+    public DosClient(URI baseUrl, String username, String password, long objectIdStartValue)
+            throws IOException {
         this.baseUrl = requireNonNull(baseUrl);
 
         String auth = username + ":" + password;
         byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
         authHeader = "Basic " + new String(encodedAuth, StandardCharsets.ISO_8859_1);
 
         httpClient = HttpClientBuilder.create().build();
+        this.objectId = objectIdStartValue;
     }
 
     public void postDataObject(Ga4ghDataObject dataObject) {
         try {
+            if (dataObject.getId() == null) {
+                dataObject.setId(Long.toString(objectId));
```

Comment:
Did you consider GUID/UUID here? If so, I'm curious what factors led you to choose stateful sequential sequencing.

Reply:
If by stateful you mean that the objectId field is in DosClient, it's a lame shortcut and it doesn't belong there. This logic should have been outside of the client. It was more convenient than trying to stuff the id generation logic inside of the for each call:
```
        objects.forEach(dosClient::postDataObject);
```
If we try to make this project cleaner some time as a whole, I'd repair that.

But anyway, fixed ing64 IDs that are preserved even between dataset reloads may be handy if you don't want to reload other related relations that use those values as foreign keys for joins.

Reply:
E.g. PGP files from the personalgenomes.ca website have fixed IDs between reloads, the 1000-genomes not necessarily, they just have file names (incl path prefix), but I synced it so it's consistent. (for the 1000genomes files I stuffed the path prefix info into the description so it is preserved somewhere)